### DISCUSSION TITLE
Can I use "Individual Apple Store Connect API key" for publishing?

### DISCUSSION THREAD

**Discussion by kele-nortal** [codemagic-ci-cd/codemagic-docs#2494] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

Upload Individual App Store Connect key, use DeveloperID as IssuerID

### Expected results

Successful publish to the store

### Actual results

Error Failed to auth

{
    "tool-version": "7.006.15006",
    "tool-path": "/Applications/Xcode-15.3.app/Contents/SharedFrameworks/ContentDeliveryServices.framework/Versions/A/Frameworks/AppStoreService.framework",
    "os-version": "14.3.1",
    "product-errors": [
        {
            "message": "Unable to upload archive.",
            "userInfo": {
                "NSLocalizedDescription": "Unable to upload archive.",
                "NSLocalizedFailureReason": "Failed to authenticate for session: (\n    \"Error Domain=ITunesConnectionAuthenticationErrorDomain Code=-26000 \\\"Failure to authenticate.\\\" UserInfo={NSLocalizedRecoverySuggestion=Failure to authenticate., NSLocalizedDescription=Failure to authenticate., NSLocalizedFailureReason=App Store operation failed.}\"\n)"
            },
            "code": -1011
        }
    ]
}

### Build id (optional)

6629007ca03d6accfd56e923

**Comment by dtrdic** [comment, marked as answer]
Hi @kele-nortal, what permission have you set for the app store connect API key?
It has to be `App Manager`. Please create another key if it has a different role then this

**Reply by kele-nortal** [reply]
Hi @dtrdic, thank you for reply.
Individual keys inherit user permission and in my case it's admin, and I cannot change it. 
So, only App Manager role then? 
Also, because I cannot test this now, setting IssuerID with DeveloperID is ok, only in my case role is wrong?

**Reply by dtrdic** [reply]
one thing that i'm noticing is that you've made the integration Apple Developer Portal under Teams > team name > team integrations > Developer portal and in there created 2 API keys, but in your yaml you are not referencing neither of them
```
   name: Android and iOS Deployment
    instance_type: mac_mini_m1
    max_build_duration: 30
    integrations:
      app_store_connect: integration // you need to add a name of one of your keys here
```
Edit: btw yes, Admin role should be fine as well

**Reply by kele-nortal** [reply]
Ok, so managed to get "Team key" with App Manager role, and now publishing works.
So there is an issue with my "Individual key" with admin role, or maybe "Individual keys" in general.
Thanks again.
<img width="1433" alt="Screenshot 2024-04-25 at 15 22 18" src="https://github.com/codemagic-ci-cd/codemagic-docs/assets/103031215/67cdab17-7fda-49e0-8f6f-e1985f7a3e35">

P.S. setting "app_store_connect" to integration is last thing I did before contacting you for help :)
